lib/route-recognizer.ts

```diff
@@ -290,26 +289,23 @@ class State {
 //  * prefers more static segments to more
 function sortSolutions(states: State[]) {
   return states.sort(function(a, b) {
-    if (!a.types) {
-      return b.types ? -1 : 0;
-    } else if (!b.types) {
-      return 1;
-    }
-    if (a.types.stars !== b.types.stars) { return a.types.stars - b.types.stars; }
+    let [ astatics, adynamics, astars ] = a.types || [0, 0, 0];
```

Comment:
using buble for that reason, to get pragmatic output, babel on the classes alone slowed down things even in loose mode, but I'm happy to change it if you'd like